backend/src/packages/achievements/achievement.entity.ts

```diff
@@ -0,0 +1,72 @@
+import { type IEntity } from '~/libs/interfaces/interfaces.js';
+import { type WithNullableKeys } from '~/libs/types/types.js';
+
+import { type AchievementEntityType } from './libs/types/types.js';
+
+class AchievementEntity implements IEntity {
+  private 'id': number | null;
+
+  private 'key': string;
+
+  private 'name': string;
+
+  private 'description': string;
+
+  private constructor({
+    id,
+    key,
+    name,
+    description,
+  }: WithNullableKeys<AchievementEntityType, 'id'>) {
+    this.id = id;
+    this.key = key;
+    this.name = name;
+    this.description = description;
+  }
+
+  public static initialize({
+    id,
+    key,
+    name,
+    description,
+  }: AchievementEntityType): AchievementEntity {
+    return new AchievementEntity({
+      id,
+      key,
+      name,
+      description,
+    });
+  }
+
+  public static initializeNew({
+    key,
+    name,
+    description,
+  }: Omit<AchievementEntityType, 'id'>): AchievementEntity {
```

Comment:
I offer move this omit to separate type

type AchievementEntityPayloadType = Omit<AchievementEntityType, 'id'>
```suggestion
  }: AchievementEntityPayloadType): AchievementEntity {
```

Reply:
Move this type before AchievmentEntity class but after imports